bolt/authorization.go

```diff
@@ -226,17 +226,20 @@ func (c *Client) CreateAuthorization(ctx context.Context, a *platform.Authorizat
 			return platform.ErrUnableToCreateToken
 		}
 
-		token, err := c.TokenGenerator.Token()
-		if err != nil {
-			return &platform.Error{
-				Err: err,
-				Op:  op,
+		if a.Token == "" {
+			token, err := c.TokenGenerator.Token()
+			if err != nil {
+				return &platform.Error{
+					Err: err,
+					Op:  op,
+				}
 			}
+			a.Token = token
 		}
-		a.Token = token
 
 		a.ID = c.IDGenerator.ID()
 
+		var err error
 		pe := c.putAuthorization(ctx, tx, a)
 		if pe != nil {
 			pe.Op = op
```

Comment:
I'd probably do `return pe` in this block and `return nil` outside it so you don't need to declare `var err error` -- slightly easier to read without the extra declaration IMO.